### PR TITLE
ENHANCE: Add automatic deploy-key addition to deploy workflow

### PR DESCRIPTION
**Pull request by paolopaolopaolo**
* Create template for `~/.yurtrc` file
* Clean up `try/except` blocks where `input` is involved
* Create method `create_yurtrc_with_access_token` in `yurt.yurt_core.deploy`
* Create method `output_git_user_access_key` in `yurt.yurt_core.deploy`
* Create method `output_var_from_base_yml` in `yurt.yurt_core.deploy`
* Make `yurt deploy` smartly find the git repo (no more errors)
* Write logic for deciding when to push SSH key to git repo deploy keys
* Fix error with PostgreSQL variable feeding

@rmutter, The deploy workflow now has an extra prompt, asking users if they'd uploaded the appropriate deploy keys yet. Wait, there's more. It then checks to see if the user has enabled an access token on GitHub yet and registered the token with Yurt (with a new nifty `~/.yurtrc` file). If there's a `~/.yurtrc` file, it then uses the access token contained within to make a POST request to the Github API to add the project's `git_ssh_pub_key` value to the repo's list of deploy keys.

### CODE REVIEW

File: yurt/yurt_core/deploy.py [hunk 1]
```diff
@@ -9,26 +17,98 @@
...
+        access_key = output_git_user_access_key()
+        # Get needed variables from base_yml
+        git_ssh_pub_key = output_var_from_base_yml('git_ssh_pub_key')
+        git_repo = output_var_from_base_yml('git_repo')
+        project_name = get_project_name_from_repo(git_repo, drop_hyphens=False)
+        owner_name = get_owner_name_from_repo(git_repo)
+        compiled_curl_command = ' '.join((
+            'curl -X POST',
+            '-H "Content-Type:application/json"',
+            '-d \'{{"title": "{}@{}", "key": "{}"}}\''.format(project_name, selection, git_ssh_pub_key),
```
Comment (rmutter) [inline yurt/yurt_core/deploy.py]: inconsistent not using indexes in string formatting here vs. using them below and elsewhere

File: yurt/yurt_core/deploy.py [hunk 2]
```diff
@@ -9,26 +17,98 @@
...
+            'https://api.github.com/repos/{}/{}/keys?access_token={}'.format(owner_name, project_name, access_key))),
```
Comment (rmutter) [inline yurt/yurt_core/deploy.py]: this works as long as you're a collaborator on an org's repo?

File: yurt/yurt_core/deploy.py [hunk 3]
```diff
@@ -9,26 +17,98 @@
...
+
+        run(compiled_curl_command)
+        run('ansible-playbook -i orchestration/inventory/{0} orchestration/site.yml'.format(selection))
```
Comment (rmutter) [inline yurt/yurt_core/deploy.py]: share this method between both bases?

File: yurt/yurt_core/deploy.py [hunk 4]
```diff
@@ -9,26 +17,98 @@
...
+
+    elif not_first_timer == 'y':
+        run('ansible-playbook -i orchestration/inventory/{0} orchestration/site.yml'.format(selection))
+
+    else:
+        print('Bad input! Try again.')
```
Comment (rmutter) [inline yurt/yurt_core/deploy.py]: maybe repeat what they inputted for ease of use?

File: yurt/yurt_core/deploy.py [hunk 5]
```diff
@@ -9,26 +17,98 @@
...
+            '4) Click on the \'Generate new token\' button.',
+            '5) Make sure this token has ALL the permissions.'
+        ])
+    )
+    github_access_key = raw_input_wrapper('\n'.join((
+       '6) Once your token has been generated, copy-paste it below and hit Enter.',
+       'Access Token: ')))
+    yurtrc_path = os.path.join(TEMPLATES_PATH, 'yurtrc.template')
+    template_vars = {'github_access_key': github_access_key}
+    run('cp {} ~/.yurtrc'.format(yurtrc_path))
```
Comment (rmutter) [inline yurt/yurt_core/deploy.py]: maybe make yurtrc path + filename a constant?

File: yurt/yurt_core/deploy.py (hunk 2 above)
Comment (paolopaolopaolo) [inline yurt/yurt_core/deploy.py]: `owner` might be a misnomer. It's assumed that the person deploying has write access to the repo, yes.

File: yurt/yurt_core/deploy.py (hunk 4 above)
Comment (paolopaolopaolo) [inline yurt/yurt_core/deploy.py]: Never! \s

**Comment by paolopaolopaolo** [comment]
@rmutter, handled all the concerns. Ready for re-review!

File: yurt/yurt_core/deploy.py [hunk 6]
```diff
@@ -71,44 +78,41 @@ def deploy():
...
-    for index, filename in enumerate(os.listdir("./orchestration/inventory")):
-        if not(filename == "vagrant"):
-            ENVIRONMENT_OPTIONS[str(index)] = filename
-            print("{0}:\t{1}".format(index, filename))
-    num_selection = raw_input_wrapper("Which environment do you want to deploy (use Option number)?:\t")
-    selection = ENVIRONMENT_OPTIONS[num_selection]
+    if selection is None:
+        print("Option\tInventory")
+        for index, filename in enumerate(os.listdir("./orchestration/inventory")):
+            if not(filename == "vagrant"):
```
Comment (rmutter) [inline yurt/yurt_core/deploy.py]: any reason this isn't just `!=`?

Comment (paolopaolopaolo) [inline yurt/yurt_core/deploy.py]: I feel like it didn't work for me in Python at one point and so I've just avoided using it. I'll try it out.

**Comment by paolopaolopaolo** [comment]
@rmutter, Fixed that last bit!

**Pull request merged**